backend/internal/service/handler/media/get_media.go

```diff
@@ -11,16 +14,36 @@ func (h *Handler) GetMediaByName(c *fiber.Ctx) error {
 }
 
 func (h *Handler) GetMedia(c *fiber.Ctx) error {
-	// Check for the "type" query parameter
-	mediaType := c.Query("sort")
-	if mediaType == "recent" {
-		// Call the method to get recent media if the "type" query parameter is "recent"
-		medias, err := h.mediaRepository.GetMediaByDate(c.Context())
+	type request struct {
+		utils.Pagination
+		Sort string `query:"sort"`
+	}
+
+	var req request
+	if err := c.QueryParser(&req); err != nil {
+		return errs.BadRequest("Invalid query parameters")
+	}
+
+	if errMap := req.Validate(); len(errMap) > 0 {
+		return errs.InvalidRequestData(errMap)
+	}
+
+	switch req.Sort {
+	case "recent":
+		media, err := h.mediaRepository.GetMediaByDate(c.Context())
 		if err != nil {
-			return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"error": "Failed to fetch recent media"})
+			return err
+		}
+
+		return c.Status(fiber.StatusOK).JSON(media)
+	case "reviews":
+		media, err := h.mediaRepository.GetMediaByReviews(c.Context(), req.Limit, req.GetOffset())
+		if err != nil {
+			return err
 		}
-		return c.Status(fiber.StatusOK).JSON(medias)
-	}
 
-	return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid type query parameter"})
+		return c.Status(fiber.StatusOK).JSON(media)
+	default:
```

Comment:
Also now that I think abt it default should prob just return page 1 limit 10 media results

What do you guys think @ddusichka @gaikwadsid

Reply:
oh yeah smart let's do that